scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -671,6 +671,38 @@ public TaskResult getTaskResultFromIncarnation(JobId jobId, String taskName, int
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @ImmediateService
+    public List<TaskResult> getTaskResultAllIncarnations(String jobId, String taskName)
+            throws NotConnectedException, UnknownJobException, UnknownTaskException, PermissionException {
+        return this.getTaskResultAllIncarnations(JobIdImpl.makeJobId(jobId), taskName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @ImmediateService
+    public List<TaskResult> getTaskResultAllIncarnations(JobId jobId, String taskName)
+            throws NotConnectedException, UnknownJobException, UnknownTaskException, PermissionException {
+        // checking permissions
+        frontendState.checkPermissions("getTaskResultFromIncarnation",
```

Comment:
we need to add getTaskResultFromIncarnation in the security.java.policy-server for the USER group

Reply:
it is there:
    permission org.ow2.proactive.permissions.MethodCallPermission "org.ow2.proactive.scheduler.core.SchedulerFrontend.getTaskResultFromIncarnation";
I didn't add a new permission, simply reused the existing one. getTaskResultAllIncarnations is simply a bulk call.